static/js/panel.js

```diff
@@ -126,4 +181,131 @@ var Panel = new (class Panel {
     this.content.setAttribute("aria-expanded", !hidden);
     this.icon.classList.toggle("expanded");
   }
+
+  copyText(copy, text) {
+    copy.setAttribute("data-copying", "true");
+    navigator.clipboard.writeText(text)
+      .then(function() {
+        copy.classList.add("copied");
+        setTimeout(function() {
+          if (!copy.hasAttribute("data-copying")) {
+            copy.classList.remove("copied");
+          }
+        }, 1000);
+      })
+      .finally(function() {
+        copy.removeAttribute("data-copying");
+      });
+  }
+
+  copyMarkdown(type) {
+    const { node, getText } = this.markdown[type];
+    if (node.classList.contains("disabled")) {
+      return;
+    }
+
+    const copy = node.querySelector(".copy");
+    const url = this.permalinkNode ? this.permalinkNode.href : document.location.href;
+    const filename = new URL(url).pathname.match(/\/([^\/]+)$/)[1];
+    const text = getText(url, filename);;
```

Comment:
We have `select_formatting` in `tools/src/languages.rs`. Seems like that could be reused by adding a `markdown_slug: &'static str` to `LanguageSpec`, and then adding the markdown-spec to the `<div id="file">` in `format_code` (maybe as a data-attribute or something), wdyt?